src/main/scala/edu/ie3/osmogrid/guardian/RunGuardian.scala

```diff
@@ -0,0 +1,256 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.osmogrid.guardian
+
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.actor.typed.scaladsl.{Behaviors, StashBuffer}
+import edu.ie3.osmogrid.cfg.OsmoGridConfig
+import edu.ie3.osmogrid.guardian.OsmoGridGuardian.Request
+import edu.ie3.osmogrid.guardian.RunGuardian.MessageAdapters.{
+  WrappedListenerResponse,
+  WrappedLvCoordinatorResponse
+}
+import edu.ie3.osmogrid.io.input.InputDataProvider
+import edu.ie3.osmogrid.io.output.ResultListener
+import edu.ie3.osmogrid.lv.LvCoordinator
+
+import java.util.UUID
+import scala.util.{Failure, Success}
+
+/** Actor to take care of a specific simulation run
+  */
+object RunGuardian extends RunSupport with StopSupport with SubGridHandling {
+  sealed trait Request
+  object Run extends Request
+
+  /** Container object with all available adapters for outside protocol messages
+    *
+    * @param lvCoordinator
+    *   Adapter for messages from [[LvCoordinator]]
+    * @param resultListener
+    *   Adapter for messages from [[ResultEventListener]]
+    */
+  private[guardian] final case class MessageAdapters(
+      lvCoordinator: ActorRef[LvCoordinator.Response],
+      resultListener: ActorRef[ResultListener.Response]
+  )
+  private[guardian] object MessageAdapters {
+    final case class WrappedLvCoordinatorResponse(
+        response: LvCoordinator.Response
+    ) extends Request
+
+    final case class WrappedListenerResponse(
+        response: ResultListener.Response
+    ) extends Request
+  }
+
+  sealed trait Response
+  final case class Done(runId: UUID) extends Response
+
+  sealed trait Watch extends Request
+  private[guardian] object InputDataProviderDied extends Watch
+
+  private[guardian] object ResultEventListenerDied extends Watch
+
+  private[guardian] object LvCoordinatorDied extends Watch
```

Comment:
This whole section implicitly creates a cross-dependency aka cyclic-dependency which I really don't like. I think we need to find another solution for this part and I may have an idea how to solve this. Please let's have a short talk.